uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/work-pattern-model.js

```diff
@@ -0,0 +1,68 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'leave-absences/shared/modules/models',
+  'leave-absences/shared/models/instances/work-pattern-instance',
+  'leave-absences/shared/apis/work-pattern-api',
+  'common/models/model'
+], function (_, models) {
+  'use strict';
+
+  models.factory('WorkPattern', [
+    '$log', 'Model', 'WorkPatternAPI', 'WorkPatternInstance',
+    function ($log, Model, workPatternAPI, instance) {
+      $log.debug('WorkPattern');
+
+      return Model.extend({
+
+        /**
+         *  Calls the assignWorkPattern() method of the WorkPattern API
+         *
+         * @param {string} contactId
+         * @param {string} workPatternID
+         * @param {string} effectiveDate
+         * @param {string} effectiveEndDate
+         * @param {string} changeReason
+         * @param {object} params - additional parameters
+         * @return {Promise}
+         */
+        assignWorkPattern: function (contactId, workPatternID, effectiveDate, effectiveEndDate, changeReason, params) {
+          return workPatternAPI.assignWorkPattern(contactId, workPatternID, effectiveDate, effectiveEndDate, changeReason, params);
+        },
+
+        /**
+         * Return the default work pattern
+         *
+         * @param  {Object} params
+         * @return {Promise}
+         */
+        default: function (params) {
```

Comment:
Why are you allowing any parameters at all for this method? The purpose for this method is, as far as i remember, to return all default work patterns, period. It shouldn't allow any "customization", it should be pretty straightforward to use
```js
WorkPattern.default().then(function (defaultPattern) {
});
```
in fact the way you wrote the implementation, you could even do this
```js
WorkPattern.default({ default: false }).then(function (nonDefaultPattern) {
});
```
basically negating the whole point of the method